api/v1alpha2/kubeadmbootstrapconfig_types.go

```diff
@@ -26,6 +27,9 @@ type KubeadmConfigSpec struct {
 	ClusterConfiguration kubeadmv1beta1.ClusterConfiguration `json:"clusterConfiguration"`
 	InitConfiguration    kubeadmv1beta1.InitConfiguration    `json:"initConfiguration,omitempty"`
 	JoinConfiguration    kubeadmv1beta1.JoinConfiguration    `json:"joinConfiguration,omitempty"`
+	// AdditionalUserDataFiles specifies extra files to be passed to user_data upon creation.
+	// +optional
+	AdditionalUserDataFiles []cloudinit.Files `json:"additionalUserDataFiles,omitempty"`
```

Comment:
I would expect to also see an update to the generated CRDs for this update.

Should we expose types from the `cloudinit` package here? I would expect others importing our types wouldn't need to also import part of the implementation.

Reply:
I'll get the CRD yaml updated (CI should fail if that's not the case!)

In CAPA we were exposing `userdata.Files` but I see your point. I'll see about an update to correct it.

Reply:
@detiber updated, ptal